spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -565,11 +589,78 @@ def _measure_tissue_bridges(self, im_lesion_data, p_lst, idx):
+        # im_lesion_data to RPI using orient2rpi())
+        # Note that this number is the same across all lesions (idx) as the midsagittal slice is based on the spinal
+        # cord segmentation
+        mid_sagittal_sc_slice = self.sagittal_sc_slice
+
+        # Get the lesion mask for the midsagittal slice
+        im_data_midsagittal = im_lesion_data[mid_sagittal_sc_slice, :, :]
+
+        # Compute the length of the lesion along the superior-inferior axis in the midsagittal slice
+        # The length is computed as the sum of the angle corrected axial slice thickness (p_lst[2])
+        # Note: p_lst[2] is the pixel size in the S-I direction
+        # Note: '[1]' in 'np.unique(np.where(im_data_midsagittal)[1])' is used to get the unique axial slices (S-I
+        # direction) with the lesion
+        length_cur = np.sum([p_lst[2] / np.cos(self.angles_sagittal[axial_slice])
+                             for axial_slice in np.unique(np.where(im_data_midsagittal)[1])])
+
+        # Save the length of the lesion along the superior-inferior axis in the midsagittal slice
+        self.measure_pd.loc[idx, 'length_midsagittal_slice [mm]'] = length_cur
+        printv('  (S-I) length in the midsagittal slice: ' + str(np.round(length_cur, 2)) + ' mm',
+               self.verbose, type='info')
+
+    def _measure_width_midsagittal_slice(self, im_lesion_data, p_lst, idx):
+        """
+        Measure the width of the lesion along the anterior-posterior axis in the **midsagittal slice** when taking into
+        account the angle correction.
+
+        :param im_lesion_data: 3D numpy array: mask of the lesion. The orientation is assumed to be RPI (because we
+        reoriented the image to RPI using orient2rpi())
+        :param p_lst: list, pixel size of the lesion
+        :param idx: int, index of the lesion
+        """
+        # Get the midsagittal slice number (the number is RPI oriented which is fine because we reoriented the
+        # im_lesion_data to RPI using orient2rpi())
+        # Note that this number is the same across all lesions (idx) as the midsagittal slice is based on the spinal
+        # cord segmentation
+        mid_sagittal_sc_slice = self.sagittal_sc_slice
+
+        # Get the lesion mask for the midsagittal slice
+        im_data_midsagittal = im_lesion_data[mid_sagittal_sc_slice, :, :]
+
+        # Compute the width of the lesion along the anterior-posterior axis in the midsagittal slice
+        # The width is computed as the sum of the angle corrected coronal slice thickness (p_lst[1])
+        # Note: 'p_lst[1]' is the pixel size in the A-P direction
+        # Note: '[0]' in ' np.unique(np.where(im_data_midsagittal)[0]' is used to get the unique coronal slices (A-P
+        # direction) with the lesion
+        width_cur = np.sum([p_lst[1] * np.cos(self.angles_sagittal[coronal_slice])
+                            for coronal_slice in np.unique(np.where(im_data_midsagittal)[0])])
```

Comment:
I think this formula for angle correction is wrong, sadly. It's because `self.angles_sagittal` is defined with indices in the Z direction (S-I axis), but `coronal_slice` is an index in the Y direction (A-P axis).

If you think about it, this is a fundamental problem: intersecting the centerline with an axial S-I slice defines a unique point, so we can talk about the centerline angle at that point. But intersecting the centerline with a coronal A-P slice may give zero or many points, so there is no well-defined angle.

Maybe the following algorithm would give an acceptable definition of the lesion width?
- Loop over all axial slices.
- If there are no lesion voxels in the slice, the A-P lesion width at this axial slice is zero.
- If there is at least one lesion voxel in the slice, the uncorrected A-P lesion width (in voxels) is `(max Y coordinate) - (min Y coordinate) + 1`. Multiply by `p_lst[1] * np.cos(self.angles_sagittal[axial_slice])` to get the angle-corrected lesion width in mm for this axial slice.
- Take the maximum over all axial slices.

You can also replace the `max-min+1` with a sum over unique indices _in the current axial slice_, like the current code does. The answer would only be different if your lesion contained a "hole" in the middle, or jagged edges. I think `max-min+1` would be more similar to what a person would do with a ruler when looking at a sagittal image, but you would know better than me about that.